### PR TITLE
[SPARK-15543][SQL] Rename DefaultSources to make them more self-describing

### PR DESCRIPTION
**Pull request by rxin**
## What changes were proposed in this pull request?

This patch renames various DefaultSources to make their names more self-describing. The choice of "DefaultSource" was from the days when we did not have a good way to specify short names.

They are now named:
- LibSVMFileFormat
- CSVFileFormat
- JdbcRelationProvider
- JsonFileFormat
- ParquetFileFormat
- TextFileFormat

Backward compatibility is maintained through aliasing.
## How was this patch tested?

Updated relevant test cases too.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala [hunk 1]
```diff
@@ -107,7 +107,7 @@ private[libsvm] class LibSVMOutputWriter(
  *  @see [[https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/ LIBSVM datasets]]
  */
 @Since("1.6.0")
-class DefaultSource extends FileFormat with DataSourceRegister {
+class LibSVMFileFormat extends FileFormat with DataSourceRegister {
```
Comment (rxin) [inline mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala]: cc @jkbradley

**Comment by marmbrus** [comment]
LGTM

File: mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala (hunk 1 above)
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala]: Can we add this to `backwardCompatibilityMap` just to be safe?

Comment (rxin) [inline mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala]: we can ...

Comment (rxin) [inline mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala]: it won't be type safe though.

Comment (rxin) [inline mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala]: i added it

**Comment by rxin** [comment]
Merging in master/2.0.

**Pull request closed without merging**